.pre-commit-config.yaml

```diff
@@ -0,0 +1,22 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.1.0
+    hooks:
+      - id: check-ast
+      - id: check-case-conflict
+      - id: check-merge-conflict
+      - id: mixed-line-ending
+        args: [--fix, lf]
+      - id: name-tests-test
+        args: [--django]
```

Comment:
`args: [--django]` looks like a copy and paste error, since all the test files are named `*Test.py` and not `test*.py` as --django assumes.

Anyhow, the default option would not work as well. See https://github.com/pre-commit/pre-commit-hooks#name-tests-test

Reply:
Basically, as long as the naming convention of test files is fixed, the testrunner can then find them using the `-p PATTERN` argument during unit test discovery. The default is `test*.py` and the `--django` option also matches `test*.py`. That was why I chose this. But the default `name-tests-test` can also be used by just providing the -p flag to the unit test command.

Reply:
We would have to potentially rename the tests once this decision is made.

Reply:
Hi,

I believe that before enabling `precommit` test themself should be changed.
As majority of tests encompass the whole learning procedure, few dozen times. And that can be quite time consuming, especially for stochastic learning. That is on a TODO list for some time now.

Reply:
You can keep the procedure tests, those can be useful, but separate them from the fast unit tests.
Then you can have both: quick checks on commit and extensive functionality tests when you feel like it or when you make a big change.

Reply:
Good idea.
It will be like that then. Now "only" thing left to do is to write unit tests 👯